plugins/blockvolume/blockprovider/gluster-block/glusterblock.go

```diff
@@ -95,45 +90,59 @@ func (g *GlusterBlock) CreateBlockVolume(name string, size uint64, hostVolume st
 	}, err
 }
 
-// DeleteBlockVolume deletes a gluster block volume of give name
-func (g *GlusterBlock) DeleteBlockVolume(name string, options ...blockprovider.BlockVolOption) error {
-	var (
-		blockVolOpts = &blockprovider.BlockVolumeOptions{}
-		hostVol      string
-	)
-
-	blockVolOpts.ApplyOpts(options...)
+// GetAndDeleteBlockVolume deletes a gluster block volume of give name
+func (g *GlusterBlock) GetAndDeleteBlockVolume(name string, options ...blockprovider.BlockVolOption) (blockprovider.BlockVolume, error) {
+	blkVol, err := g.GetBlockVolume(name)
+	if err != nil || blkVol == nil {
+		return nil, errors.ErrBlockVolNotFound
+	}
+	err = g.deleteBlock(name, blkVol, options...)
+	if err != nil {
+		log.WithError(err).Errorf("couldn't delete block volume :%s", name)
+		return nil, err
+	}
 
-	blockVols := g.BlockVolumes()
+	return blkVol, nil
 
-	for _, blockVol := range blockVols {
-		if blockVol.Name() == name {
-			hostVol = blockVol.HostVolume()
-			break
-		}
-	}
+}
 
-	if hostVol == "" {
-		return errors.New("block volume not found")
+// DeleteBlockVolume deletes a gluster block volume of give name
+func (g *GlusterBlock) DeleteBlockVolume(name string, options ...blockprovider.BlockVolOption) error {
```

Comment:
this function and `GetAndDeleteBlockVolume` looks similar, any major difference, can we avoid code duplication?